### PR TITLE
Work lohoum squalene lat lng to meters

### PR DESCRIPTION
**Pull request by loh0um**
We created the class to project latlng from a sphere into x,y coordinates on a plane in meters. This will be useful since the creation of the mapping mission was done in meters and we acquire the data by letting the user click on mapbox, which uses latlng. We also refactored the code to give an easier interface to use.

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt [hunk 1]
```diff
@@ -1,87 +1,22 @@
-/*
- * Copyright (C) 2021  Drone3D-Team
- * The license can be found in LICENSE at root of the repository
- */
-
```
Comment (GabrielFleischer) [inline app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt:5]: Is removing the copyright intended ?

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt [hunk 2]
```diff
@@ -1,87 +1,22 @@
...
 
 import ch.epfl.sdp.drone3d.drone.DroneService
-import javax.inject.Inject
+import com.mapbox.mapboxsdk.geometry.LatLng
 
-class MappingMissionService @Inject constructor(private val droneService: DroneService) {
+interface MappingMissionService {
 
-    private val cameraAngle = 0.0 // Suppose the drone is looking down
+    val droneService:DroneService
```
Comment (GabrielFleischer) [inline app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt:8]: Why is the service needed here ?

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/Parallelogram.kt [hunk 3]
```diff
@@ -8,12 +8,13 @@ package ch.epfl.sdp.drone3d.mission
...
 
 /**
- * Utility class to represent a parallelogram
+ * Utility class to represent a parallelogram in meters
  */
 data class Parallelogram(val origin:Point, val dir1Span:Vector, val dir2Span:Vector){
 
     constructor(origin: Point,adjacentPoint1:Point,adjacentPoint2:Point): this(origin,Vector(origin,adjacentPoint1),Vector(origin,adjacentPoint2))
 
+
```
Comment (GabrielFleischer) [inline app/src/main/java/ch/epfl/sdp/drone3d/mission/Parallelogram.kt:17]: I think this line is unnecessary

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ */
+
+package ch.epfl.sdp.drone3d.mission
+
+import ch.epfl.sdp.drone3d.drone.DroneService
+import com.mapbox.mapboxsdk.geometry.LatLng
+import java.lang.IllegalArgumentException
+import javax.inject.Inject
+
+class ParallelogramMappingMissionService @Inject constructor(override val droneService: DroneService):MappingMissionService{
```
Comment (GabrielFleischer) [inline app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt:13]: The style is not perfect, there should be a space after the :

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    private fun buildMappingMission(
+        vertices: List<LatLng>,
+        flightHeight: Double,
+        mappingFunction: (
+            startingPoint: Point, area: Parallelogram, cameraAngle: Double,
+            flightHeight: Double, groundImageDimension: GroundImageDim
+        ) -> List<Point>
+    ): List<LatLng>? {
+
+        assert(vertices.size==3)
```
Comment (GabrielFleischer) [inline app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt:43]: Assert should not be used in android. Prefere if (...) thrown IllegalArguementException()

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            null
+        }
+    }
+
+    /**
+     * Returns the dimension of the projected image to the ground using [flightHeight] in meters
+     * and the camera properties provided by the drone
+     * If a property is not provided by the drone, returns null
+     */
+
```
Comment (GabrielFleischer) [inline app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt:63]: This line is unnecessary

**Review by Anghul — APPROVED** [review]

**Review by GabrielFleischer — CHANGES_REQUESTED** [review]
The MappingMissionService interface has an unnecessary field.
And there are some style issues

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt (hunk 6 above)
Comment (Squalene) [reply app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt]: Corrected

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt (hunk 5 above)
Comment (Squalene) [reply app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt]: Corrected

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt (hunk 4 above)
Comment (Squalene) [reply app/src/main/java/ch/epfl/sdp/drone3d/mission/ParallelogramMappingMissionService.kt]: Corrected

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/Parallelogram.kt (hunk 3 above)
Comment (Squalene) [reply app/src/main/java/ch/epfl/sdp/drone3d/mission/Parallelogram.kt]: Corrected

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt (hunk 2 above)
Comment (Squalene) [reply app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt]: Corrected

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt (hunk 1 above)
Comment (Squalene) [reply app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt]: Corrected

**Review by GabrielFleischer — APPROVED** [review]

**Pull request merged**